speechbrain/tokenizers/SentencePiece.py

```diff
@@ -484,3 +486,53 @@ def get_spm_tokens(model_path):
     model.load(model_path)
     mapping = [model.sp.id_to_piece(i) for i in range(model.sp.vocab_size())]
     return mapping
+
+
+@dataclass
+class SentencePieceDecoderStreamingContext:
+    """Mutable streaming context for a single SentencePiece streaming session.
+    """
+
+    emitted_symbol_count: int = 0
+    """The number of symbols that have been emitted for this transcription."""
+
+
+def spm_decode_preserve_leading_space(
+    tokenizer: spm.SentencePieceProcessor,
+    hyps: List[int],
+    context: SentencePieceDecoderStreamingContext,
+) -> List[str]:
```

Comment:
the -> operator makes streaming incompatible with older version of python, is that really ok? @mravanelli @Adel-Moumen

Reply:
I think it's fine. We did it on others places as well and we didn't had any issues with it.